### PR TITLE
8249879: Split MemorySegment and MemoryAddress (jextract edition)

### PR DESCRIPTION
**Pull request by mcimadamore**
This patch fixes uses of memory segments and address following the big refactoring introduced by https://git.openjdk.java.net/panama-foreign/pull/260 from sources and tests.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8249879](https://bugs.openjdk.java.net/browse/JDK-8249879): Split MemorySegment and MemoryAddress ⚠️ Title mismatch between PR and JBS.

### Reviewers
 * Athijegannathan Sundararajan ([sundar](@sundararajana) - Committer)

### Download
`$ git fetch https://git.openjdk.java.net/panama-foreign pull/267/head:pull/267`
`$ git checkout pull/267`

### CODE REVIEW

File: src/jdk.incubator.jextract/share/classes/jdk/internal/clang/Cursor.java [hunk 1]
```diff
@@ -335,26 +335,15 @@ public Cursor getSpecializedCursorTemplate() {
         return new Cursor(Index_h.clang_getSpecializedCursorTemplate(cursor));
     }
```
Comment (sundararajana) [inline src/jdk.incubator.jextract/share/classes/jdk/internal/clang/Cursor.java:337]: This diff removes the callback cache fix for jextract performance. I believe this is unintentional merge artefact.

File: src/jdk.incubator.jextract/share/classes/jdk/internal/clang/Utils.java [hunk 2]
```diff
@@ -27,46 +27,27 @@
...
     }
 
-    static void setInt(MemoryAddress addr, int i) {
-        INT_VH.set(addr, i);
+    static void setLong(MemorySegment addr, long i) {
+        MemoryAccess.setLong(addr, i);
     }
 
-    static int getLong(MemoryAddress addr) {
-        return (int)LONG_VH.get(addr);
```
Comment (sundararajana) [inline src/jdk.incubator.jextract/share/classes/jdk/internal/clang/Utils.java:53]: Do we need this renaming wrapper? BTW most of this Utils class is just forwarding to MemoryAccess. Might as well use MemoryAccess directly in callsties?

Comment (mcimadamore) [reply src/jdk.incubator.jextract/share/classes/jdk/internal/clang/Utils.java:53]: We could, yes

**Pull request closed without merging**